client/clients/tso/dispatcher.go

```diff
@@ -371,11 +371,15 @@ func (td *tsoDispatcher) handleProcessRequestError(
 	default:
 	}
 
-	// Release this stream from the manager due to error.
-	conCtxMgr.Release(streamURL)
 	// Update the member list to ensure the latest topology is used before the next batch.
 	svcDiscovery := td.provider.getServiceDiscovery()
+	// Release this stream from the manager due to error.
+	conCtxMgr.Release(streamURL)
+	if errs.ShouldRedial(err) {
+		svcDiscovery.RemoveClientConn(streamURL)
+	}
 	if errs.IsLeaderChange(err) {
+		metrics.TSOLeaderChangedCounter.Inc()
```

Comment:
What if this err is `NotPrimaryErr`?

Reply:
Do you mean the naming issue?

Reply:
No, I mean this counter should not include `notLeaderError` also `NotPrimaryErr`